projects/power_distribution/src/front_uv_detector.c

```diff
@@ -0,0 +1,32 @@
+#include "can_transmit.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress s_uv_comp_pin_address = { .port = GPIO_PORT_B, .pin = 0 };
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+static void prv_pin_interrupt_handler(const GpioAddress *address, void *context) {
+  CAN_TRANSMIT_UV_CUTOFF_NOTIFICATION();
+
+  LOG_WARN("UV lockout was triggered\n");
+}
+
+void prv_register_interrupts(void) {
```

Comment:
Since this is a private function, it should be `static` - otherwise there could be an obscure error if another file had a `prv_register_interrupts`.